package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1362 @@
+       *
+       * When set to an empty string or a _nully_ value, the attribute value is _reset_.
+       *
+       * When reset, the attribute assumes its _default value_
+       * (except on the top-root type, `Instance.type`, in which case it has no effect).
+       *
+       * The _default value_ is the _inherited value_.
+       *
+       * The _initial value_ of the attribute on the top-root type is `"instance"`.
+       *
+       * @type {!nonEmptyString}
+       */
+      get label() {
+        return this._label;
+      },
+
+      set label(value) {
+        value = nonEmptyString(value);
+        if(value === null) {
+          this._resetLabel();
+        } else {
+          this._labelSet = true;
+          this._label = value;
+        }
+      },
+
+      _resetLabel: function() {
+        this._labelSet = false;
+
+        if(this !== _type) {
+          delete this._label;
+        }
+      },
+      //endregion
+
+      //region description property
+
+      // -> nonEmptyString, Optional, Inherited, Configurable, Localized
+      // "" -> null conversion
+
+      _description: null, // set through implement bundle, below
+
+      /**
+       * Gets or sets the description of this type.
+       *
+       * Attempting to set to a non-string value type implicitly converts the value to a string before assignment.
+       *
+       * Setting to `undefined` causes this attribute to use the inherited value,
+       * except for the root type, `Instance.type` (which has no ancestor), where this attribute is `null`.
+       *
+       * Setting to `null` or to an empty string clears the attribute and sets it to `null`,
+       * ignoring any inherited value.
+       *
+       * @type {?nonEmptyString}
+       */
+      get description() {
+        return this._description;
+      },
+
+      set description(value) {
+        if(value === undefined) {
+          this._resetDescription();
+        } else {
+          this._description = nonEmptyString(value);
+        }
+      },
+
+      _resetDescription: function() {
+        if(this !== _type) {
+          delete this._description;
+        }
+      },
+      //endregion
+
+      //region category property
+
+      // -> nonEmptyString, Optional, Inherited, Configurable, Localized
+      // "" -> null conversion
+
+      _category: null,
+
+      /**
+       * Gets or sets the category associated with this type.
+       *
+       * The category is used primarily to group similar types (or instances of) in a user interface.
+       *
+       * Attempting to set to a non-string value type implicitly
+       * converts the value to a string before assignment.
+       *
+       * Setting to `undefined` causes this attribute to use the inherited value,
+       * except for the root type, `Instance.type` (which has no ancestor), where the attribute is `null`.
+       *
+       * Setting to `null` or to an empty string clears the attribute and sets it to `null`,
```

Comment:
Suggest:

Setting this to `null` or to an empty string clears the attribute and sets it to `null`,